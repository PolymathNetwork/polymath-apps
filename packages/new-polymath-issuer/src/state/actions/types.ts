import { ActionType } from 'typesafe-actions';
import * as sessionActions from './session';
import * as appActions from './app';

export type SessionActions = ActionType<typeof sessionActions>;
export type AppActions = ActionType<typeof appActions>;
