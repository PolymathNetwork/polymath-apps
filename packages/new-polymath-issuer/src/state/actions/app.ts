import { createStandardAction } from 'typesafe-actions';

export const setActiveTransactionGroup = createStandardAction(
  'APP/SET_ACTIVE_TRANSACTION_GROUP'
)<string>();

export const initializePolyClientStart = createStandardAction(
  'APP/INITIALIZE_POLY_CLIENT_START'
)();

export const initializePolyClientSuccess = createStandardAction(
  'APP/INITIALIZE_POLY_CLIENT_SUCCESS'
)();

export const initializePolyClientFailure = createStandardAction(
  'APP/INITIALIZE_POLY_CLIENT_FAILURE'
)<Error>();
