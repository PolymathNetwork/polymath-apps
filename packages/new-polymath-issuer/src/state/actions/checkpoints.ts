import { types } from '@polymathnetwork/new-shared';
import { createEntityActions } from '~/state/helpers/createEntityActions';

const { createAction, updateAction, deleteAction } = createEntityActions<
  types.CheckpointEntity
>('CHECKPOINTS');

export { createAction, updateAction, deleteAction };
