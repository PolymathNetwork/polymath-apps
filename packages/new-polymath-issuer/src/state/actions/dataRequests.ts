import { createStandardAction } from 'typesafe-actions';
import { RequestKeys, Fetcher } from '~/types';
import { types } from '@polymathnetwork/new-shared';

const invalidateRequest = createStandardAction('DATA_REQUESTS/INVALIDATE')<{
  requestKey: RequestKeys;
  args?: types.Pojo;
}>();

const cacheData = createStandardAction('DATA_REQUESTS/CACHE')<{
  requestKey: RequestKeys;
  args: types.Pojo;
  fetchedIds: string[];
}>();

/**
 * This action signals the saga in charge of fetching data from the blockchain
 * that a component needs certain data
 */
const requestData = createStandardAction('DATA_REQUESTS/REQUEST')<{
  fetcher: Fetcher;
}>();

const fetchDataStart = createStandardAction('DATA_REQUESTS/FETCH_START')<{
  requestKey: RequestKeys;
  args: types.Pojo;
}>();

const fetchDataFail = createStandardAction('DATA_REQUESTS/FETCH_FAIL')<{
  errorMessage: string;
  requestKey: RequestKeys;
  args: types.Pojo;
}>();

export {
  invalidateRequest,
  cacheData,
  requestData,
  fetchDataStart,
  fetchDataFail,
};
