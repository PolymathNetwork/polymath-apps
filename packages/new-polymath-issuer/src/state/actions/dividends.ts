import { types } from '@polymathnetwork/new-shared';
import { createEntityActions } from '~/state/helpers/createEntityActions';

const { createAction, updateAction, deleteAction } = createEntityActions<
  types.DividendEntity
>('DIVIDENDS');

export { createAction, updateAction, deleteAction };
