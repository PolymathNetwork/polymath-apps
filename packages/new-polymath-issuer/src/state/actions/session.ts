import { createStandardAction } from 'typesafe-actions';
import { Wallet, Identity } from '~/types';

export const setWallet = createStandardAction('SESSION/SET_WALLET')<Wallet>();
export const setIdentity = createStandardAction('SESSION/SET_IDENTITY')<
  Identity
>();
