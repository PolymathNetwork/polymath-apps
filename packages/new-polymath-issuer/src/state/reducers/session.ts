import { Reducer } from 'redux';
import { getType } from 'typesafe-actions';
import { Wallet, Identity } from '~/types';
import { SessionActions } from '~/state/actions/types';
import * as actions from '~/state/actions/session';

export interface SessionState {
  wallet?: Wallet;
  identity?: Identity;
}
const initialState: SessionState = {};

export const reducer: Reducer<SessionState, SessionActions> = (
  state = initialState,
  action
) => {
  switch (action.type) {
    case getType(actions.setIdentity): {
      return {
        ...state,
        identity: action.payload,
      };
    }
    case getType(actions.setWallet): {
      return {
        ...state,
        wallet: action.payload,
      };
    }
    default: {
      return { ...state };
    }
  }
};
