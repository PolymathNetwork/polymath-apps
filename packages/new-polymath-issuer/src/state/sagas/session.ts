import { takeLatest, all } from 'redux-saga/effects';
import { getType } from 'typesafe-actions';
import { loginStart } from '~/state/actions/session';
import { requireWallet } from './accessControl';

export function* processLogin() {
  yield requireWallet();
}

export function* sessionWatcher() {
  yield all([takeLatest(getType(loginStart), processLogin)]);
}
