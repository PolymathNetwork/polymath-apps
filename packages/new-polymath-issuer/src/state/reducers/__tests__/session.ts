import { reducer } from '../session';
import * as actions from '~/state/actions/session';
import { Identity } from '~/types';

describe('Reducer: session', () => {
  test('initialState', () => {
    expect(reducer(undefined, {} as any)).toMatchSnapshot();
  });

  test('setIdentity updates identity ', () => {
    const identity: Identity = {
      email: 'some@email',
      fullName: 'Rafael Vidaurre',
    };
    const result = reducer(undefined, actions.setIdentity(identity));
    expect(result.identity).toEqual(identity);
  });

  test("setWallet updates the session's wallet ", () => {
    const wallet = {
      address: '0x1234',
    };
    const result = reducer(undefined, actions.setWallet(wallet));
    expect(result.wallet).toEqual(wallet);
  });
});
