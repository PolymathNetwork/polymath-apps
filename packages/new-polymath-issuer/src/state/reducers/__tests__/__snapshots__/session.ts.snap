// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer: session initialState 1`] = `Object {}`;
